Reducers/FavoritesReducer.swift

```diff
@@ -0,0 +1,209 @@
+import Foundation
+import ComposableArchitecture
+
+struct FavoriteRow: Identifiable, Equatable {
+    let id: String
+    let name: String
+    let imageURL: String?
+    let breed: CatBreed
+    var isFavorite: Bool
+}
+
+@Reducer
+struct FavoritesFeature {
+    // MARK: - Navigation Route
+    @Reducer
```

Comment:
You can replace this @Reducer by @Reducer(state: .equatable, action: .equatable) and use a enum with just one case here:
```suggestion
enum Route {
            case breedDetail(BreedDetailFeature)
}
```